Cabal/Distribution/Simple/Utils.hs

```diff
@@ -1412,8 +1412,10 @@ shortRelativePath from to =
 dropExeExtension :: FilePath -> FilePath
 dropExeExtension filepath =
   case splitExtension filepath of
-    (filepath', extension) | extension `elem` exeExtensions -> filepath'
-                           | otherwise                      -> filepath
+    (filepath', extension)
+      | extension `elem` exts -> filepath'
+      | otherwise             -> filepath
+  where exts = [ "." ++ ext | ext <- exeExtensions ]
 
 -- | List of possible executable file extensions on the current platform.
 exeExtensions :: [String]
```

Comment:
Could we improve the haddocks here to say

> on the build platform

as current is ambiguous, could mean target platform too.

That would explain why `Ghcjs` entry doesn't (and shouldn't?) have `jsexe` listed.